cmd/internal/persist_output.go

```diff
@@ -22,10 +27,61 @@ into a versionsed secret or kube native secret using flags specified to this com
 			return errors.Errorf("persist-output command failed. cf-operator-namespace flag is empty.")
 		}
 
-		return extendedjob.PersistOutput(namespace)
+		outputVolumePath := viper.GetString("output-volume-path")
+		if len(outputVolumePath) == 0 {
+			return errors.Errorf("persist-output command failed. cf-operator-namespace flag is empty.")
+		}
+
+		// hostname of the container is the pod name in kubernetes
+		podName, err := os.Hostname()
+		if err != nil {
+			return errors.Wrapf(err, "failed to fetch pod name.")
+		}
+		if podName == "" {
+			return errors.Wrapf(err, "pod name is empty.")
+		}
+
+		// Authenticate with the cluster
+		clientSet, versionedClientSet, err := authenticateInCluster()
+		if err != nil {
+			return err
+		}
+
+		po := extendedjob.NewPersistOutputInterface(namespace, podName, clientSet, versionedClientSet, outputVolumePath)
+
+		return po.PersistOutput()
 	},
 }
 
 func init() {
 	utilCmd.AddCommand(persistOutputCmd)
+
+	persistOutputCmd.Flags().StringP("output-volume-path", "", "/mnt/quarks", "path to the volume mount")
```

Comment:
While we assume that an extendedjob will place STDOUT into `/mnt/quarks`, making this configurable here, will not match the actual path where the ExtendedJob generates the file. Trying to achieve this will be more complex, so I guess it will be better to remove this flag, and stick to the default `/mnt/quarks/ `directory.

Reply:
Done. Removed.